mypy/build.py

```diff
@@ -840,197 +648,6 @@ def stats_summary(self) -> Mapping[str, object]:
         return self.stats
 
 
-# Package dirs are a two-tuple of path to search and whether to verify the module
-OnePackageDir = Tuple[str, bool]
-PackageDirs = List[OnePackageDir]
-
-
-class FindModuleCache:
-    """Module finder with integrated cache.
-
-    Module locations and some intermediate results are cached internally
-    and can be cleared with the clear() method.
-
-    All file system accesses are performed through a FileSystemCache,
-    which is not ever cleared by this class. If necessary it must be
-    cleared by client code.
-    """
-
-    def __init__(self, fscache: Optional[FileSystemCache] = None,
-                 options: Optional[Options] = None) -> None:
-        self.fscache = fscache or FileSystemCache()
-        # Cache find_lib_path_dirs: (dir_chain, search_paths) -> list(package_dirs, should_verify)
-        self.dirs = {}  # type: Dict[Tuple[str, Tuple[str, ...]], PackageDirs]
-        # Cache find_module: (id, search_paths, python_version) -> result.
-        self.results = {}  # type: Dict[Tuple[str, SearchPaths, Optional[str]], Optional[str]]
-        self.options = options
-
-    def clear(self) -> None:
-        self.results.clear()
-        self.dirs.clear()
-
-    def find_lib_path_dirs(self, dir_chain: str, lib_path: Tuple[str, ...]) -> PackageDirs:
-        # Cache some repeated work within distinct find_module calls: finding which
-        # elements of lib_path have even the subdirectory they'd need for the module
-        # to exist. This is shared among different module ids when they differ only
-        # in the last component.
-        # This is run for the python_path, mypy_path, and typeshed_path search paths
-        key = (dir_chain, lib_path)
-        if key not in self.dirs:
-            self.dirs[key] = self._find_lib_path_dirs(dir_chain, lib_path)
-        return self.dirs[key]
-
-    def _find_lib_path_dirs(self, dir_chain: str, lib_path: Tuple[str, ...]) -> PackageDirs:
-        dirs = []
-        for pathitem in lib_path:
-            # e.g., '/usr/lib/python3.4/foo/bar'
-            dir = os.path.normpath(os.path.join(pathitem, dir_chain))
-            if self.fscache.isdir(dir):
-                dirs.append((dir, True))
-        return dirs
-
-    def find_module(self, id: str, search_paths: SearchPaths,
-                    python_executable: Optional[str]) -> Optional[str]:
```

Comment:
FWIW `python_executable` is never actually used by `find_module`, `_find_module`, and `find_module_recursive`, it just gets passed around but I didn't find a place where it is used. As I understand, all the info is stored already in `search_paths` that are calculated using the executable.

Reply:
Yes, that's why I removed it from their signatures in the refactor.